lib/sales_analyst.rb

```diff
@@ -159,4 +159,103 @@ def invoice_status(status)
     all = sales_engine.invoices_per_status[status]
     ((all / total_invoices.to_f) * 100).round(2)
   end
+
+  def invoice_paid_in_full?(invoice_id)
+    successes = sales_engine.find_all_by_result(:success)
```

Comment:
Could change successes to invoices_with_status_succeed to be more clear

Reply:
good idea